test/TesterInternal/RuntimeCodeGenTests.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Threading.Tasks;
+using Orleans;
+using Orleans.ApplicationParts;
+using Orleans.Hosting;
+using Orleans.Metadata;
+using Orleans.Serialization;
+using Xunit;
+
+namespace UnitTests
+{
+    [TestCategory("BVT"), TestCategory("CodeGen")]
+    public class RuntimeCodeGenTests
+    {
+        [Fact]
+        public void RuntimeCodeGenAddsSupportClasses()
```

Comment:
Is it worth adding a functional that starts a silo and verifies the grain and serialization works?

Reply:
Sure thing, will do